rmf_robot_sim_common/include/rmf_robot_sim_common/slotcar_common.hpp

```diff
@@ -126,15 +126,16 @@ class SlotcarCommon
   {
     double v = 0.0; // Target displacement in X (forward)
     double w = 0.0; // Target displacement in yaw
-    double speed = 0.0; // Target speed at next waypoint
+    double target_linear_speed_now = 0.0;     // Target linear speed now
+    double target_linear_speed_destination = 0.0; // Target linear speed at destination
     std::optional<double> max_speed = std::nullopt; // Maximum speed allowed while navigating
   };
 
+  // Ignition plugin reads this to display markers
   struct PursuitState
   {
-    double lookahead_radius = 0.0;
     Eigen::Vector3d lookahead_point = Eigen::Vector3d::Zero();
-    std::vector<Eigen::Vector3d> lookahead_segments;
+    std::size_t traj_wp_idx;
```

Comment:
I noticed that the `traj_wp_idx` seems to be unused on the Ignition side, maybe it can be removed? If it can we could even get rid of the whole `PursuitState` struct since it would only encapsulate a single variable?

Reply:
Yea these can be removed, done.